### ISSUE TITLE
Add 'free' item to cart

### ISSUE THREAD

**Issue by Anne-Thraks** [wojodesign/simplecart-js#192]
Is it possible to add an item to the basket automatically when the total exceeds a certain value?

ie. a free gift is added to the basket when the total spend meets a certain amount.

(the item would also have to be removed if the total dropped below that same value)

**Comment by brettwejrowski** [comment]
You can use the afterAdd event to check the total and if the item is already there, then add the item accordingly

**Comment by Anne-Thraks** [comment]
```
simpleCart.bind( 'afterAdd' , function( item ){ 
    if( simpleCart.total > 30){
        simpleCart.add( 'name=Free Item' , 'price=0' , 'quantity=1' , 'thumb=freeitem_thumb.jpg' );
        }
    return false;
});
```

Thanks for the advice, I've tried the above but it keeps looping through, continually adding Free Items as soon as the cart value exceeds 30.
Any thoughts on how I can get it to add (or remove) the free item only once depending on total value?

**Comment by brettwejrowski** [comment]
You will want to check the name of the item being added:

``` javascript

simpleCart.bind( 'afterAdd' , function( item ){ 
    if( simpleCart.total > 30 && item.name !== "Free Item" ){
        simpleCart.add( 'name=Free Item' , 'price=0' , 'quantity=1' , 'thumb=freeitem_thumb.jpg' );
        }
    return false;
});

```

**Comment by Anne-Thraks** [comment]
Thanks for that, however this seems to add a free item 'EACH' time I add an item (when the total is above 30) rather than adding 'ONLY ONE' free Item when the total is above 30.

I'm also stumped on how to remove the 'Free Item' if the total subsequently drops below 30, I can't seem to trigger anything to happen when the cart total drops below 30 or a decrement occurs in the cart.

**Comment by brettwejrowski** [comment]
Makes sense.  Instead of using the 'afterAdd' event, you might want to tie into the update function, which would be called whenever items are removed or added.  Then you can just add or remove the free item there (Don't forget to check if its already there).  You can iterate through the items to see if it has the free item by using the simpleCart.each() function.

**Comment by Anne-Thraks** [comment]
Thanks for the fast response, presumably this would mean editing the simplecart.js file or is there a way to tie into the update function without editing the code?

My solution:

```
me.update = function() {
    ...
    me.save();
    me.freeItem(); // call new function at end of update function
};

me.freeItem = function() {
    if ( me.total >= 30 && typeof me.items['Free_Item']=='undefined' ) {
        // if total is high enough & the free item does not exist then add one...
        me.add( 'name=Free Item' , 'price=0' , 'quantity=1' , 'thumb=freeitem_thumb.jpg' , 'id=Free_Item' );
    }
    else if ( me.total < 30 && me.items['Free_Item'].quantity > 0 ) {
        // if total is not high enough & there is a free item in the cart, remove it...
        me.items['Free_Item'].remove();
    }
}
```

Hope this helps someone.

**Comment by adelphiaUK** [comment]
Couldn't you add the free item "before checkout?"
